go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -300,28 +301,44 @@ func (npw *nodePortWatcher) updateServiceFlowCache(service *corev1.Service, netI
 				// Flows that will DNAT all traffic coming into nodeport to the nodeIP:Port and
 				// ensure that the return traffic is UnDNATed to correct the nodeIP:Nodeport
 				if isServiceTypeETPLocal && hasLocalHostNetworkEp {
+					// Get the target port (this is important for named ports).
+					svcPortKey := util.GetServicePortKey(svcPort.Protocol, svcPort.Name)
+					lbe, err := localEndpoints.GetLBEndpoints(svcPortKey)
+					if err != nil {
+						errors = append(errors, err)
+						continue
+					}
+					targetPort := lbe.Port
+
 					// case1 (see function description for details)
 					var nodeportFlows []string
-					klog.V(5).Infof("Adding flows on breth0 for Nodeport Service %s in Namespace: %s since ExternalTrafficPolicy=local", service.Name, service.Namespace)
+					klog.V(5).Infof("Adding flows on breth0 for Nodeport Service %s with targetPort %d in "+
+						"Namespace: %s since ExternalTrafficPolicy=local", service.Name, targetPort, service.Namespace)
 					// table 0, This rule matches on all traffic with dst port == NodePort, DNAT's the nodePort to the svc targetPort
 					// If ipv6 make sure to choose the ipv6 node address for rule
 					if strings.Contains(flowProtocol, "6") {
+						// Regardless of the IP addresses in the local endpoints list, we always send to npw.gatewayIPv6.
```

Comment:
This comment alarmed me quite a bit. Can we rephrase it? :-D   
Service ports are orthogonal to the IP family in use and localEndpoints just includes the port information for our practical purposes. Is this what you wanted to convey? :)